pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -377,4 +378,7 @@ type FrameworkHandle interface {
 
 	// GetWaitingPod returns a waiting pod given its UID.
 	GetWaitingPod(uid types.UID) WaitingPod
+
+	// Clientset returns a kubernetes clientset.
+	Clientset() clientset.Interface
```

Comment:
We can't provide all the lister supported by kubernetes, but the developer could create lister or informer by themselves with the client set IMO.

Reply:
The new listers/informers do not share memory with old ones. For big clusters, it will waste a lot of memory.

Reply:
> The new listers/informers do not share memory with old ones. For big clusters, it will waste a lot of memory.

The kubernetes clientset is necessary for creating and updating resources, and we should be careful of exposing variables in the framework handle. I think now that the developer could create lister with the clientset, it would be better for us to add them when the performance issue happens, which proved to be a bottleneck.

Reply:
Yes, we could expose interface in the framework handle.

> it would be better for us to add them when the performance issue happens, which proved to be a bottleneck.

It is not performance issue, it wastes a lot memory.

Reply:
> It is not performance issue, it wastes a lot memory.

Sorry, what I mean is we should add them when the problem happens since the kubernetes clientset would meet most of the requirements for the plugins.

Reply:
Do you have the numbers of memory consumption in a big cluster? If it is indeed a problem, we should open an issue for that.

Reply:
I haven't thought deeply about this. But a rule of thumb is to use SharedInformerFactory instead of individual informers.

Reply:
Yeah, the informer is one of the common use cases here. But we also need to use the client to communicate to the API server directly, e.g. update the resources. 

https://github.com/kubernetes/kubernetes/blob/8aca2c148018770e4ad0f4b21d98a81ff714f731/plugin/pkg/scheduling/defaultbinder/plugin.go#L43-L56

Reply:
ClientSet is something some plugins will need to update/write resources, but we certainly shouldn't recommend that individual plugins create their own informers for two reasons: 1) the most common pattern is to create a single SharedInformerFactory to avoid memory and communication overhead, 2) for a some specific resources (nodes, pods), the plugins should list from the scheduler cache.

Having said that, it is not necessary to make this part of the FrameworkHandle and officially make it part of the API. A Plugin can get the clientset, sharedinformerfactory or listers at the time it is instantiated and stores references to the ones it cares about as local variables in the plugin struct.

Reply:
Thinking about this a little more, we will need to make it part of the FrameworkHandle so that out-of-tree plugins have access, so we can't just rely on capturing those at instantiation time